multiqc/modules/snpsplit/snpsplit.py

```diff
@@ -0,0 +1,202 @@
+        self.allele_sorting_section()
+
+    def parse_new_snpsplit_log(self, f):
+        data = next(yaml.load_all(f['f'], Loader=yaml.SafeLoader))
+        flat_data = {}
+        for k in data:
+            for sk in data[k]:
+                newkey = '{}_{}'.format(k.lower(), sk)
+                flat_data[newkey] = data[k][sk]
+        input_fn = data['Meta']['infile']
+        return [input_fn, flat_data]
+
+    def parse_old_snpsplit_log(self, f):
+        input_fn = None
+        d = dict()
+
+        for line in f['f'].splitlines():
+            # Parse the sample name
+            match = re.match(r"Input file:\W+'([^']+)'", line)
+            if match:
+                # Report can have two sections with two input files,
+                # the second sorting the output from the tagging
+                # Just take the first instance, which is the initial input file
+                if input_fn is None:
+                    input_fn = match.group(1)
+                continue
+
+            regex_patterns = [
+                # Allele-tagging report
+                ['tagging_skipped', r"Reads were unaligned and hence skipped: (\d+)"],
+                ['tagging_unassignable', r"(\d+) reads were unassignable"],
+                ['tagging_genome1', r"(\d+) reads were specific for genome 1"],
+                ['tagging_genome2', r"(\d+) reads were specific for genome 2"],
+                ['tagging_unassignableCT', r"(\d+) reads that were unassignable contained C>T SNPs"],
+                ['tagging_ambiguous', r"(\d+) reads did not contain one of the expected bases"],
+                ['tagging_conflictingSNPs', r"(\d+) contained conflicting allele-specific SNPs"],
+                # Allele-specific sorting report
+                ['sorting_conflictingReads', r"Reads contained conflicting SNP information:\W+(\d+)"]
+            ]
+            for (k, regex) in regex_patterns:
+                match = re.match(regex, line)
+                if match:
+                    d[k] = int(match.group(1))
+                    break
+
+            # Allele-specific sorting report
+            sorting_patterns = [
+                ['sorting_unassignableReads', "Reads were unassignable"],
+                ['sorting_unassignableReads', "Read pairs were unassignable (UA/UA):"],
+                ['sorting_genome1Reads', "Reads were specific for genome 1"],
+                ['sorting_genome1Reads', "Read pairs were specific for genome 1 (G1/G1)"],
+                ['sorting_genome2Reads', "Reads were specific for genome 2"],
+                ['sorting_genome2Reads', "Read pairs were specific for genome 2 (G2/G2)"],
+                ['sorting_G1UA', "Read pairs were a mix of G1 and UA"],
+                ['sorting_G2UA', "Read pairs were a mix of G2 and UA"],
+                ['sorting_G1G2', "Read pairs were a mix of G1 and G2"],
+                ['sorting_conflictingReads', "Read pairs contained conflicting SNP information"]
```

Comment:
Need to do the same matching up here...

* `sorting_unassignableReads`
* `sorting_genome1Reads`
* `sorting_genome2Reads`
* `sorting_G1UA`
* `sorting_G2UA`
* `sorting_G1G2`
* `sorting_conflictingReads`

New fields:

* `sorting_PE_conflicting`
* `sorting_PE_conflicting_pairs`
* `sorting_PE_conflicting_singletons`
* `sorting_PE_genome1`
* `sorting_PE_genome1_pairs`
* `sorting_PE_genome1_singletons`
* `sorting_PE_genome2`
* `sorting_PE_genome2_pairs`
* `sorting_PE_genome2_singletons`
* `sorting_PE_total_pairs`
* `sorting_PE_total_reads`
* `sorting_PE_total_singletons`
* `sorting_PE_unassignable`
* `sorting_PE_unassignable_pairs`
* `sorting_PE_unassignable_singletons`

If this gets too difficult / complicated I guess we could also just ditch support for the old-style logs.. How long were they around for? How important is it to maintain support for them?

Reply:
I suppose it is fair enough to say that MultiQC integrations works from the YAML file onwards. Wouldn't you agree?

Reply:
I'd feel more comfortable saying that if SNPsplit had any stable releases that generated YAML files 😅 (or any documentation about the different fields)

Reply:
I would feel more comfortable to make a stable release if I knew that MultiQC would be fine with format... :) How about I will make one as soon as I see the first beautiful plots? Need to attend a campfire in the garden right now, talk later.

Reply:
Hah, fair enough.. I'll see what I can do 👍

Reply:
Rules for understanding: trim `PE_` / `SE_` prefix, ignore anything that is `_pairs` or `_singletons` (key without suffix is the total). Then we have fewer fields to match and it gets easier:

* `sorting_PE_conflicting`
* `sorting_PE_genome1`
* `sorting_PE_genome2`
* `sorting_PE_total_pairs`
* `sorting_PE_unassignable`